### PR TITLE
Added a download button

### PR DESCRIPTION
**Pull request by ozearkhan**
### Summary

This pull request introduces a new feature to the js-beautify project: a "Download Code" button that allows users to download the beautified code in the format they choose (HTML, CSS, or JS). 

Fixes Issue: 
- #2309 

### Changes Made

1. **Added Download Button to `index.html`**
   - A new button labeled "Download Beautified Code" has been added to the HTML.
   - Clicking this button triggers the `downloadBeautifiedCode` function.

2. **Implemented `downloadBeautifiedCode` Function**
   - Added a function in `web/common-function.js` to handle the download functionality.
   - The function generates a Blob from the beautified code and initiates a download with a filename based on the selected language.

#### `index.html`

- **New Button Added:**
  ```html
  <button class="control" type="button" onclick="downloadBeautifiedCode()"><strong>Download Beautified Code</strong></button>
  ```

#### `web/common-function.js`

- **New Function: `downloadBeautifiedCode`**
  ```javascript
  function downloadBeautifiedCode() {
    var output = the.lastOutput; // The beautified code
    var language = $('#language').val();
    var filename = 'beautified_code.' + (language === 'html' ? 'html' : language === 'css' ? 'css' : 'js');
    
    var blob = new Blob([output], { type: 'text/' + (language === 'html' ? 'html' : language) });
    var link = document.createElement('a');
    link.href = URL.createObjectURL(blob);
    link.download = filename;
    document.body.appendChild(link);
    link.click();
    document.body.removeChild(link);
  }
  ```

### Testing

- **Manual Testing:**
  - Verified that the "Download Beautified Code" button appears and functions correctly.
  - Ensured that clicking the button downloads the beautified code in the correct format and with the correct filename.
  

### Additional Information

- **Screenshots:**
  
Screenshot 2024-09-11 131114

Screenshot 2024-09-11 131140

- **Related Issues:**
  - No issues found.

---

### CODE REVIEW

**Comment by ozearkhan** [comment]
@bitwiseman  sir this PR is ready for review.

File: web/common-function.js [hunk 1]
```diff
@@ -150,6 +150,38 @@ function unpacker_filter(source) {
...
+  // Creating a temporary anchor element to trigger the download
+  var link = document.createElement("a");
+  link.href = URL.createObjectURL(blob);
+  link.download = "js-beautify." + fileExtension; // Dynamic file name based on extension
+
+  // Triggering the download
+  link.click();
+
+  // Cleanup
+  URL.revokeObjectURL(link.href);
```
Comment (bitwiseman) [inline web/common-function.js:182]: Does this need a try-finally around it? 
```suggestion
  link.href = URL.createObjectURL(blob);
  try {
    link.download = "js-beautify." + fileExtension; // Dynamic file name based on extension

    // Triggering the download
    link.click();
  } finally {
    // Cleanup
    URL.revokeObjectURL(link.href);
  }
```

File: index.html [hunk 2]
```diff
@@ -99,7 +99,9 @@ <h1 class="logo">
       <button class="control" type="button" onclick="copyText()"><strong>Copy to Clipboard</strong></button>
       <button class="control" type="button" onclick="selectAll()"><strong>Select All</strong></button>
       <button class="control" type="button" onclick="clearAll()"><strong>Clear</strong></button>
+      <button class="control" type="button" onclick="downloadBeautifiedCode()"><strong>Download Code</strong></button>
       <input type="file" onchange="changeToFileContent(this)">
+
```
Comment (bitwiseman) [inline index.html:104]:
```suggestion
```

File: web/common-function.js (hunk 1 above)
Comment (ozearkhan) [reply web/common-function.js:182]: yeah my bad I should have wrapped it around with try , finally .

**Comment by ozearkhan** [comment]
Sir do you need any other changes?

File: web/common-function.js [hunk 3]
```diff
@@ -150,6 +150,40 @@ function unpacker_filter(source) {
...
+  }
+
+  // Creating a Blob object with the content
+  var blob = new Blob([content], { type: "text/plain;charset=utf-8" });
+
+  // Creating a temporary anchor element to trigger the download
+  var link = document.createElement("a");
+  link.href = URL.createObjectURL(blob);
+  try {
+    link.download = "js-beautify." + fileExtension; // Dynamic file name based on extension
```
Comment (bitwiseman) [inline web/common-function.js:177]:
```suggestion
    link.download = "beautified." + fileExtension; // Dynamic file name based on extension
```

**Review by bitwiseman — APPROVED** [review]

**Pull request merged**